### DISCUSSION TITLE
ImportError libGL when using CV2

### DISCUSSION THREAD

**Discussion by KnightCoder** [Project-MONAI/monai-deploy-app-sdk#456] [Q&A]
If the MONAI Deploy SDK app uses CV2 then it throws an error `ImportError: libGL.so.1: cannot open shared object file: No such file or directory` . I think having a way to execute Linux commands on Docker image creation would've helped (similar to DOCKERFILE). 

MONAI Deploy SDK version: 0.5.1
Referring doc: https://docs.monai.io/projects/monai-deploy-app-sdk/en/0.5.1/developing_with_sdk/packaging_app.html

**Comment by MMelQin** [comment]
@KnightCoder Thanks for the raising this discussion.

Indeed, supporting installing system packages (Debian packages) is important for Python packages that dependent on such. There had been an issue on the board for quite awhile, though at the moment, the approach would be to first create a container image based on the default one that is used by the Packager, installing the needed library as the application calls for, then on packaging the application into MAP, pass the the app specific container image as the base image.

**Reply by KnightCoder** [reply]
However, since MONAI Deploy package's base image parameter only accepts image names / tags from rocm or  nvcr.io, I am unable to run this as well. Read here https://docs.monai.io/projects/monai-deploy-app-sdk/en/0.5.1/developing_with_sdk/packaging_app.html#example-of-using-monai-application-packager.

**Reply by MMelQin** [reply]
Sorry if I did not make it clear enough.
This would be a two step process.

1. Built a container image based on the default base image that the packager uses, call it say my_base:1.0. The needed packages are installed in this image.
2. Run the packager with the newly created container image, my_base:1.0, as the base image, using the argument on the command line.

With this approach, one can first verify the custom base image works properly before using it as the base image for a MAP.
